### ISSUE TITLE
Bug: ["@typescript-eslint/array-type"] suggests incorrect type when specifying "default"

### ISSUE THREAD

**Issue by iisaduan** [typescript-eslint/typescript-eslint#9498]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVJxwB4BReEADwxFWAGd4AlEKYPCATyLoxi1QDmAbQC6APjEAKSrygAueCQCUCkgG4AsAChtQA&eslintrc=N4KABBYEQA4DYFcDmBLAdgZygLjAbXEgigAEAXATxgFMMBjAJxRjIFpa50ypCIBdADS9oDBHFo4woIsXJVajZmw5cA9AEMGDdRVaUakgjMhRqWgPYMoxosKJCbU6ABNqAM3VjuuKJu26MFABbeGprAF87MD5hSPCgA&tsconfig=&tokens=false

### Repro Code

```TypeScript
declare function foo<E extends Readonly<string[]>>(extra: E): E;
```

### ESLint Config

```javascript
{
    "plugins": [
      "@typescript-eslint"
    ],
    "rules": {
      "@typescript-eslint/array-type": [
        "error, { "default": "array-simple" }
      ]
    }
}
```

### Expected Result

Expected error message: `Array type using 'Readonly<string[]>' is forbidden for simple types. Use 'readonly string[]' instead.`

### Actual Result

The actual error message is incorrect--`Array type using 'Readonly<string[]>' is forbidden for simple types. Use 'readonly string[][]' instead. 1:32 - 1:50` It suggests a type that is not the same as the written type.

### Additional Info

I encountered these while looking though the CI tests for DefinitelyTyped. All of the cases have the same extra `[]` in error message regardless of the type used (not just `string`-- the wrong message also happens with generic types, type parameters, and constant/defined types). 

Removing the `default` part of the config and just specifying `"@typescript-eslint/array-type": [ "error" ]` gives the expected error message.

**Comment by Josh-Cena** [comment]
The error message is wrong but the autofix is correct.
